src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolBoundHandle.WindowsThreadPool.cs

```diff
@@ -162,7 +165,7 @@ private static unsafe void OnNativeIOCompleted(IntPtr instance, IntPtr context,
             ThreadPool.IncrementCompletedWorkItemCount();
 
             if (NativeRuntimeEventSource.Log.IsEnabled())
-                NativeRuntimeEventSource.Log.ThreadPoolIODequeue((NativeOverlapped*)overlappedPtr);
+                NativeRuntimeEventSource.Log.ThreadPoolIODequeue(Win32ThreadPoolNativeOverlapped.ToNativeOverlapped(overlapped));
```

Comment:
This event should be raised before CompleteWithCallback above (before the user's callback is invoked)